testing/yield.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Install "yield" support to Mocha tests.
+ * Check test-yield.js for how-to.
+ */
+export function installYieldIt(realIt) {
+  it = enableYield.bind(null, realIt); // eslint-disable-line no-native-reassign, no-undef
```

Comment:
we can do either:
```
it.yields('message', function*() {
});
```

or

```
it('message', function*() {
});
```

I'm in favor of the latter because it makes the other syntax like `it.only` and `it.skip` unchanged.